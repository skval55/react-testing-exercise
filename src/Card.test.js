import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";

// smoke test
it("renders without crashing", function() {
  render(
    <Card caption={"test"} src={"testimg.jpg"} currNum={1} totalNum={1} />
  );
});

// snapshot test
it("matches snapshot", () => {
  const { asFragment } = render(
    <Card caption={"test"} src={"testimg.jpg"} currNum={1} totalNum={1} />
  );
  expect(asFragment()).toMatchSnapshot();
});
