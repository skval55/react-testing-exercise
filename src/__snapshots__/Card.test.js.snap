// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      test
    </h4>
    <img
      alt="test"
      class="Card-image"
      src="testimg.jpg"
    />
    <small
      class="Card-small"
    >
      Image 1 of 1.
    </small>
  </div>
</DocumentFragment>
`;
